pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -526,16 +526,37 @@ def system_prompt(self, func: Callable[[], str], /) -> Callable[[], str]: ...
     @overload
     def system_prompt(self, func: Callable[[], Awaitable[str]], /) -> Callable[[], Awaitable[str]]: ...
 
+    @overload
+    def system_prompt(
+        self, func: _system_prompt.SystemPromptFunc[AgentDeps] | None = None, /, *, dynamic: bool = False
```

Comment:
I think this overload can't have `func`, so the signature is just `(self, *, dynamic: bool = False)`

Reply:
I've taken out the entire overload, it was not useful here.